### ISSUE TITLE
Readings...

### ISSUE THREAD

**Issue by jakezp** [mletenay/home-assistant-goodwe-inverter#13]
Hey,

Not really sure this is an issue, probably more just me not grasping the values being exposed by the sensors...

It'll be great if you can help me out making sense of these values.

<img width="470" alt="Screenshot 2020-10-16 at 22 18 01" src="https://user-images.githubusercontent.com/6917873/96307218-8c500c00-1001-11eb-87c4-925fd1d6731f.png">

So, I've been reading through all the resolved issues, etc. trying to get a better understanding and I understand that sensor.house_consumption =  sensor.ppv - sensor.pbattery1 + sensor.pgrid. And also that sensor.pload + sensor.gback_up = sensor.plant_power. This part makes sense to me.

But should sensor.plant_power not also be the same (or close to) sensor.house_consumption? as per the attached screenshot, sensor.house_consumption = 598W and sensor.plant_power is 807W. With sensor.pbattery1 = 598W, sensor.pgrid = 0W and sensor.ppv also = 0W. So if the plant total really does = 807W, where is the difference of 209W coming from, if not from the battery? What am I missing here?

And just to confuse me a bit more, there is another total in there, sensor.total_power, which does not match anything else. What does sensor.total_power represent?

Any pointers on helping me clear up the confusion?

The other thing that is confusing me is that my total load and generation for the day is being reported as sensor.e_load_day = 13.2kWh (SEMS Portal - 15.29kWh) and sensor.e_day = 9.5kWh (SEMS Portal - 11.40kWh). Is there something I am missing with the calculation between what's reported on SEMS Portal vs the inverter directly? I'd understand it being slightly out, but this is quite significant.

![sems](https://user-images.githubusercontent.com/6917873/96307454-f1a3fd00-1001-11eb-8d56-7be8d78d6796.png)

Thanks.

**Comment by jakezp** [comment]
Update...

After the initial post last night night (at 22:50), I've picked up that sensor.e_day has increased from 9.5kWh to 10.2kWh between 22:50 and 23:32, which should really not be possible, since it's the middle of the night and no PV generation.

<img width="411" alt="Screenshot 2020-10-17 at 21 52 16" src="https://user-images.githubusercontent.com/6917873/96352446-7a36a200-10c3-11eb-823d-6175316384cb.png">

I'm now dumping the data to MQTT every 10 seconds instead of feeding it directly into HA. From here I'm feeding the data into EmonCMS. I'll collect data for a 24 hour period and then compare the totals in kWh for each of the sensors and see how it compares to the totals reported on SEMS portal and to sensor.e_day and sensor.e_load_day.

I'm still not much closer understanding how house_consumption, plant_power and total_power plays together.

Thanks...

**Comment by jakezp** [comment]
I've been able to collect some more data on e_day values. I've been logging everything to MQTT since last night and posting the data to EmonCMS, I've been able to put the following together.

<img width="1130" alt="Screenshot 2020-10-18 at 15 54 31" src="https://user-images.githubusercontent.com/6917873/96369501-a05b5100-115a-11eb-9a3d-e683a63b3f37.png">

Yellow - e_day
Red - actual ppv readings
Blue - ppv kWh calculated based on ppv readings

Check e_day between 00:00 - 01:28... It keeps stepping up with 100W every few minutes (00:04, 00:16, 00:29, 00:41, 00:52, 01:05, 01:17 & 01:28), then it's stable until 07:57.

Also, comparing e_day and ppv_kwh it definitely seems it's not capturing all the data points. At 15:52, e_day is sitting at 6kWh and ppv_kWh is at 8.4kWh. SEMS Portal is reporting also reporting 8.4kWh at more or less the same time.

So either the way the inverter is saving e_day is completely off, or the way it's being read (Unless I'm doing something wrong)...
